core/src/main/scala/package.scala

```diff
@@ -40,4 +40,15 @@ package object sfs extends sfs.api.Api {
       case Failure(t) => l(t)
     }
   }
+
+  trait Functor[F[_]] {
+    def map[A, B](f: A => B): F[A] => F[B]
+  }
+  object Functor {
+    implicit def forMappable[A, M[A] <: { def map[B](f: A => B): M[B] }]: Functor[M] =
```

Comment:
Oh, I was responding to your comments, I'm on a phone it's hard to read code. Structural types are terrible in practice, there's no two ways about it. If someone wants that implicit they can write it, but we don't. I'd use code generation before structural types.